test/unit_tests/test_workflow.py

```diff
@@ -453,3 +458,113 @@ def whatever():
         pass
 
     assert whatever.__doc__ == "Do whatever."
+
+
+def test_step_group_basic():
+    @step("Sub step 1")
+    def sub_step1(n):
+        return {"x": n + 1}
+
+    @step("Sub step 2")
+    def sub_step2(n: int, x: int):
+        return {"x": x * n}
+
+    @step("Sub step 3")
+    def sub_step3(n: int, x: int):
+        return {"x": x + n}
+
+    group = step_group("Multiple steps", begin >> sub_step1 >> sub_step2 >> sub_step3)
+
+    wf = workflow("Workflow with step group step")(lambda: init >> step1 >> step2 >> group >> step3 >> done)
+
+    log = []
+
+    pstat = create_new_process_stat(wf, {"n": 3})
+    result = runwf(pstat, store(log))
+    assert_complete(result)
+    assert [t[0] for t in log] == [
+        "Start",
+        "Step 1",
+        "Step 2",
+        "Sub step 1",
+        "Sub step 2",
+        "Sub step 3",
+        "Multiple steps",
+        "Step 3",
+        "Done",
+    ]
+
+    def _state_key_fn(p: Process, k: str) -> Any:
+        return p.unwrap().get(k)
+
+    assert [_state_key_fn(t[1], "__step_group") for t in log] == [None] * 3 + ["Multiple steps"] * 3 + [None] * 3
+    assert [_state_key_fn(t[1], "__sub_step") for t in log] == [None] * 3 + [
+        "Sub step 1",
+        "Sub step 2",
+        "Sub step 3",
+    ] + [None] * 3
+
+
+def test_step_group_with_inputform_suspend():
+    @step("Sub step")
+    def sub_step(name: str):
+        return {"name_validate": name}
+
+    group = step_group("Multistep", begin >> step2 >> user_action >> sub_step)
+
+    wf = workflow("Workflow with step group step")(lambda: init >> step1 >> group >> step3 >> done)
+
+    log = []
+
+    pstat = create_new_process_stat(wf, {})
+    result = runwf(pstat, logstep=store(log))
+
+    assert_suspended(result)
+    assert log == [
+        ("Start", Success({})),
+        ("Step 1", Success({"steps": [1]})),
+        ("Step 2", Success({"steps": [1, 2], "__sub_step": "Step 2", "__step_group": "Multistep"})),
+        ("Input Name", Suspend({"steps": [1, 2], "__sub_step": "Input Name", "__step_group": "Multistep"})),
+        ("Multistep", Suspend({"steps": [1, 2], "__sub_step": "Input Name", "__step_group": "Multistep"})),
+    ]
+
+
+def test_step_group_with_inputform_resume():
```

Comment:
I think it is usefull to create a test where you make the input step fail

Reply:
Good point